proto/proto-backplane-grpc/src/main/proto/deephaven/proto/table.proto

```diff
@@ -579,6 +592,162 @@ message ComboAggregateRequest {
+    optional bool average_evenly_divided = 1;
+  }
+
+  message AggSpecPercentile {
+    // The percentile to calculate. Must be in the range [0.0, 1.0].
+    double percentile = 1;
+
+    // Whether to average the highest low-bucket value and lowest high-bucket value, when the low-bucket and high-bucket
+    // are of equal size. Only applies to numeric types.
+    optional bool average_evenly_divided = 2;
+  }
+
+  message AggSpecSorted {
+    // Using a message instead of string to support backwards-compatibility in the future
+    repeated AggSpecSortedColumn columns = 1;
+  }
+
+  message AggSpecSortedColumn {
+    // TODO(deephaven-core#821): SortedFirst / SortedLast aggregations with sort direction
+    string column_name = 1;
+  }
+
+  message AggSpecTDigest {
+    // T-Digest compression factor. Must be greater than or equal to 1. Defaults to 100. 1000 is extremely large.
+    optional double compression = 1;
+  }
+
+  message AggSpecUnique {
+    // Whether to include null values as a distinct value for determining if there is only one unique value to output.
+    optional bool include_nulls = 1;
+    optional AggSpecNonUniqueSentinel non_unique_sentinel = 2;
+  }
+
+  message AggSpecNonUniqueSentinel {
+    oneof type {
+      string string_value = 1;
+      sint32 int_value = 2;
+      sint64 long_value = 3 [jstype=JS_STRING];
+      float float_value = 4;
+      double double_value = 5;
+      bool bool_value = 6;
+    }
+  }
+
+  message AggSpecWeighted {
+    // Column name for the source of input weights.
+    string weight_column = 1;
+  }
+
+  oneof type {
+      AggSpecBlank abs_sum = 1;
+      AggSpecApproximatePercentile approximate_percentile = 2;
+      AggSpecBlank avg = 3;
+      AggSpecCountDistinct count_distinct = 4;
+      AggSpecDistinct distinct = 5;
+      AggSpecBlank first = 6;
+      AggSpecFormula formula = 7;
+      AggSpecBlank freeze = 8;
+      AggSpecBlank group = 9;
+      AggSpecBlank last = 10;
+      AggSpecBlank max = 11;
+      AggSpecMedian median = 12;
+      AggSpecBlank min = 13;
+      AggSpecPercentile percentile = 14;
+      AggSpecSorted sorted_first = 15;
+      AggSpecSorted sorted_last = 16;
+      AggSpecBlank std = 17;
+      AggSpecBlank sum = 18;
+      AggSpecTDigest t_digest = 19;
+      AggSpecUnique unique = 20;
+      AggSpecWeighted weighted_avg = 21;
+      AggSpecWeighted weighted_sum = 22;
+      AggSpecBlank var = 23;
+  }
+}
+
+message AggregationRequest {
+  Ticket result_id = 1;
+  TableReference source_id = 2;
+  optional TableReference initial_groups_id = 3;
+  optional bool preserve_empty = 4;
+  repeated Aggregation aggregations = 5;
+  repeated string group_by_columns = 6;
+}
+
+message Aggregation {
+  message AggregationColumns {
+    AggSpec spec = 1;
+    repeated string match_pairs = 2;
+  }
+
+  message AggregationCount {
+    // The output column name
+    string column_name = 1;
+  }
+
+  message AggregationRowKey {
```

Comment:
@niloc132 how do you feel about this sort of thing? Should we be defining separate messages?